scripts/assign-colors.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+"""
+Generate colors.tsv for augur export based on ordering, color schemes, and what exists in the metadata
+"""
+import argparse
+import pandas as pd
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Assign colors based on ordering",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument('--ordering', type=str, required=True, help="input ordering file")
+    parser.add_argument('--color-schemes', type=str, required=True, help="input color schemes file")
+    parser.add_argument('--metadata', type=str, help="if provided, restrict colors to only those found in metadata")
+    parser.add_argument('--output', type=str, required=True, help="output colors tsv")
+    args = parser.parse_args()
+
+    assignment = {}
+    with open(args.ordering) as f:
+        for line in f.readlines():
+            array = line.lstrip().rstrip().split("\t")
+            if len(array) == 2:
+                name = array[0]
+                trait = array[1]
+                if name not in assignment:
+                    assignment[name] = [trait]
+                else:
+                    assignment[name].append(trait)
+
+    # if metadata supplied, go through and
+    # 1. remove assignments that don't exist in metadata
+    # 2. remove assignments that have 'focal' set to 'False' in metadata
+    if args.metadata:
+        metadata = pd.read_csv(args.metadata, delimiter='\t')
+        for name, trait in assignment.items():
+            if name in metadata:
+                subset_present = [x for x in assignment[name] if x in metadata[name].unique()]
+                assignment[name] = subset_present
+            if name in metadata and 'focal' in metadata:
+                focal_list = metadata.loc[metadata['focal'], name].unique()
+                subset_focal = [x for x in assignment[name] if x in focal_list]
+                assignment[name] = subset_focal
```

Comment:
When the condition on L41 is true, this code is doing the work on L39-40 and then throwing it away. Suggest re-ordering the logic is that only one subset is built and assigned:

```suggestion
            if name in metadata:
                if 'focal' in metadata:
                    focal_list = metadata.loc[metadata['focal'], name].unique()
                    subset_focal = [x for x in assignment[name] if x in focal_list]
                    assignment[name] = subset_focal
                else: # no 'focal' present
                    subset_present = [x for x in assignment[name] if x in metadata[name].unique()]
                    assignment[name] = subset_present

```